google/api_core/retry.py

```diff
@@ -89,9 +94,55 @@ def check_if_exists():
 _ASYNC_RETRY_WARNING = "Using the synchronous google.api_core.retry.Retry with asynchronous calls may lead to unexpected results. Please use google.api_core.retry_async.AsyncRetry instead."
 
 
+class RetryFailureReason(Enum):
```

Comment:
Ok, I did some reorganization, moving the files into their own `retry` folder. The TYPE_CHECKING was a bit more complicated, since import * doesn't play nicely with our linting (and the generics don't exist in the retry_base) , but I think it still lets us simplify things a bit

I also made sure to keep existing exported names the same, to leave existing code unchanged.

Let me know what you think